### DISCUSSION TITLE
size in feature.xml files, is it good for anything?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/p2#213] [General]
Currently a feature xml get these default attributes for a plugin:

```
<plugin
         id="org.eclipse.jdt.core"
         download-size="0"
         install-size="0"
         version="0.0.0"
         unpack="false"/>
```

An Tycho computes and updates the size during the build. This can lead to baseline problems detected because the "content" of the feature is different than before (currently there is only a warning that is most of the time ignored).

Given that P2 has its own meta-data I wonder how useful this values are of today and if probably Tycho should just delete these attributes and PDE do no longer add them by default so we end up with simply:

```
<plugin id="org.eclipse.jdt.core"/>
```

**Comment by mickaelistria** [comment]
I think it was useful before p2 to implement some progress report. Nowadays, I'm pretty sure we could live without it.

**Comment by vogella** [comment]
+1, only unpack is used IIRC, also by Tycho I think

**Reply by laeubi** [reply]
> +1, only unpack is used IIRC, also by Tycho I think

Not even that, see
 - https://github.com/eclipse-tycho/tycho/issues/1035

**Comment by vogella** [comment]
+2 for removal is this case

**Comment by HannesWell** [comment]
Looks reasonable to remove the sizes and the unpack.
PDE should then be adjusted to remove those attributes and should guide users how to replace the unpack attribute.
